react-components/src/hooks/useSearchMappedEquipmentFDM.tsx

```diff
@@ -50,55 +50,81 @@ export const useSearchMappedEquipmentFDM = (
   return useQuery(
     ['reveal', 'react-components', 'search-mapped-fdm', query, models, viewsToSearch],
     async () => {
-      const viewsWithVersion = await createSourcesFromViews(viewsToSearch, fdmSdk);
-
       if (models.length === 0 || viewsToSearch.length === 0) {
         return [];
       }
 
-      if (query === '') {
-        const result = await fdmSdk.queryNodesAndEdges(
-          createMappedEquipmentQuery(models, viewsWithVersion, limit)
-        );
-
-        const transformedResults = convertQueryNodeItemsToSearchResultsWithViews(
-          result.items.mapped_nodes.concat(result.items.mapped_nodes_2) as NodeItem[]
+      const sources = await createSourcesFromViews(viewsToSearch, fdmSdk);
+      const chunkedSources = chunk(sources, 10);
+
+      const queryResults: SearchResultsWithView[] = [];
+
+      for (const sourceChunk of chunkedSources) {
+        queryResults.push(
+          ...(await searchNodesWithViewsAndModels(
+            query,
+            spacesToSearch,
+            sourceChunk,
+            models,
+            instancesFilter,
+            fdmSdk,
+            limit
+          ))
         );
+      }
+      return queryResults;
+    },
+    { staleTime: Infinity }
+  );
+};
 
-        const combinedWithOtherViews = viewsWithVersion.map((view) => ({
-          view,
-          instances:
-            transformedResults.find(
-              (result) =>
-                result.view.externalId === view.externalId && result.view.space === view.space
-            )?.instances ?? []
-        }));
+const searchNodesWithViewsAndModels = async (
+  query: string,
+  spacesToSearch: string[],
+  sourcesToSearch: Source[],
+  models: AddModelOptions[],
+  instancesFilter: any,
+  fdmSdk: FdmSDK,
+  limit: number = 100
+): Promise<SearchResultsWithView[]> => {
+  if (query === '') {
+    const result = await fdmSdk.queryNodesAndEdges(
+      createMappedEquipmentQuery(models, sourcesToSearch, limit)
+    );
 
-        return combinedWithOtherViews;
-      }
+    const transformedResults = convertQueryNodeItemsToSearchResultsWithViews(
+      result.items.mapped_nodes.concat(result.items.mapped_nodes_2) as NodeItem[]
+    );
 
-      const searchResults: SeachResultsWithView[] = [];
+    const combinedWithOtherViews = sourcesToSearch.map((view) => ({
+      view,
+      instances:
+        transformedResults.find(
```

Comment:
Can we use `reduce` here!

Reply:
Discussed this, constructing a map for lookup seems a bit unnecessary here, the list will have at most 10 elements.